File: releases-hub-gradle-plugin/src/main/java/com/releaseshub/gradle/plugin/task/ListDependenciesTask.java
package com.releaseshub.gradle.plugin.task;


import com.releaseshub.gradle.plugin.common.AbstractTask;

import org.gradle.api.logging.LogLevel;


public class ListDependenciesTask extends AbstractTask {

	public ListDependenciesTask() {
		setDescription("List all dependencies");
	}

	@Override
	protected void onExecute() {
		getLogger().log(LogLevel.LIFECYCLE, "Hello world");
	}

}

File: releases-hub-gradle-plugin/src/main/java/com/releaseshub/gradle/plugin/task/ListDependenciesTask.kt
package com.releaseshub.gradle.plugin.task

import com.releaseshub.gradle.plugin.common.AbstractTask
import com.releaseshub.gradle.plugin.dependencies.BasicDependenciesExtractor

open class ListDependenciesTask : AbstractTask() {

    companion object {
        const val TASK_NAME = "listDependencies"
    }

    init {
        description = "List all dependencies"
    }

    override fun onExecute() {

        val extractor = BasicDependenciesExtractor(getAllDependenciesPaths())
        val dependenciesParserResult = extractor.extractArtifacts(project.rootProject.projectDir, includes, excludes)

        getAllDependenciesPaths().forEach {
            val artifacts = dependenciesParserResult.artifactsMap[it]
            log(it)
            if (artifacts.isNullOrEmpty()) {
                log(" - No dependencies found here")
            } else {
                artifacts.forEach { artifact ->
                    log(" - $artifact:${artifact.fromVersion}")
                }
            }
            log("")
        }
    }
}
